internal/ls/findallreferences.go

```diff
@@ -0,0 +1,1493 @@
+		// 	// In the case where we're looking inside comments/strings, we don't have
+		// 	// an actual definition.  So just use 'undefined' here.  Features like
+		// 	// 'Rename' won't care (as they ignore the definitions), and features like
+		// 	// 'FindReferences' will just filter out these results.
+		// 	state.addStringOrCommentReference(sourceFile.FileName, createTextSpan(position, search.text.length));
+		// }
+
+		return
+	}
+
+	if getMeaningFromLocation(referenceLocation)&state.searchMeaning == 0 {
+		return
+	}
+
+	referenceSymbol := state.checker.GetSymbolAtLocation(referenceLocation)
+	if referenceSymbol == nil {
+		return
+	}
+
+	parent := referenceLocation.Parent
+	if parent.Kind == ast.KindImportSpecifier && parent.PropertyName() == referenceLocation {
+		// This is added through `singleReferences` in ImportsResult. If we happen to see it again, don't add it again.
+		return
+	}
+
+	if parent.Kind == ast.KindExportSpecifier {
+		// !!! not implemented
+		// debug.Assert(referenceLocation.Kind == ast.KindIdentifier || referenceLocation.Kind == ast.KindStringLiteral)
+		// state.getReferencesAtExportSpecifier(referenceLocation /* Identifier | StringLiteral*/, referenceSymbol, parent.AsExportSpecifier(), search, addReferencesHere, false /*alwaysGetReferences*/)
+		return
+	}
+
+	// !!! not implemented
+	// if isJSDocPropertyLikeTag(parent) && parent.isNameFirst &&
+	// 	parent.TypeExpression && isJSDocTypeLiteral(parent.TypeExpression.Type()) &&
+	// 	parent.TypeExpression.Type().jsDocPropertyTags && length(parent.TypeExpression.Type().jsDocPropertyTags)  {
+	// 	getReferencesAtJSDocTypeLiteral(parent.TypeExpression.Type().jsDocPropertyTags, referenceLocation, search, state);
+	// 	return;
+	// }
+
+	relatedSymbol, relatedSymbolKind := state.getRelatedSymbol(search, referenceSymbol, referenceLocation)
+	if relatedSymbol == nil {
+		state.getReferenceForShorthandProperty(referenceSymbol, search)
+		return
+	}
+
+	switch state.specialSearchKind {
+	case "none":
+		if addReferencesHere {
+			state.addReference(referenceLocation, relatedSymbol, relatedSymbolKind)
+		}
+	case "constructor":
+		// !!! not implemented
+		// state.addConstructorReferences(referenceLocation, sourceFile, search)
+	case "class":
+		// !!! not implemented
+		// state.addClassStaticThisReferences(referenceLocation, search)
+	}
+
+	// Use the parent symbol if the location is commonjs require syntax on javascript files only.
+	if ast.IsInJSFile(referenceLocation) && referenceLocation.Parent.Kind == ast.KindBindingElement &&
+		ast.IsVariableDeclarationInitializedToRequire(referenceLocation.Parent.Parent.Parent) {
+		// !!! when findAllReferences has been fully implemented, check if the behavior is the same since isVariableDeclarationInitializedToBareOrAccessedRequire has been removed
```

Comment:
only the tests that use a property access on a require should fail, eg `const x = require('mod').x`